pkg/snapshot/restoreclient.go

```diff
@@ -314,28 +315,42 @@ func (o *RestoreClient) createRestoreRequest(ctx context.Context, vConfig *confi
 }
 
 func (o *RestoreClient) skipKey(key string) bool {
-	if !o.vConfig.PrivateNodes.Enabled {
+	if o.vConfig.PrivateNodes.Enabled {
+		// check if the snapshot exists
+		if strings.HasPrefix(key, pvcPrefix) {
+			pvcName := strings.TrimPrefix(key, pvcPrefix)
+			status, ok := o.snapshotRequest.Status.VolumeSnapshots.Snapshots[pvcName]
+			if !ok {
+				return false
+			}
+			// skip restoring PVC if it has a snapshot, restore controller will restore it
+			return status.Phase == volumes.RequestPhaseCompleted
+		} else if strings.HasPrefix(key, pvPrefix) {
+			volumeName := strings.TrimPrefix(key, pvPrefix)
+			for _, snapshotSpec := range o.snapshotRequest.Spec.VolumeSnapshots.Requests {
+				if snapshotSpec.PersistentVolumeClaim.Spec.VolumeName == volumeName {
+					return true
+				}
+			}
+		}
+	} else {
 		// Restore skips restoring PVs and PVCs only in private mode.
 		// In the shared mode, PVs and PVCs are restored in virtual cluster, and volumes are restored in the host
```

Comment:
This comment should now be updated (part of it should also be moved to the if block for private nodes), because it changed for shared nodes, as now with shared nodes it also skips PVCs for which the snapshots have failed.

Reply:
sure. Will make these changes.